debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java

```diff
@@ -195,6 +201,42 @@ public synchronized void start(Map<String, String> props) {
         }
     }
 
+  /**
+   * Calculate the set of new tables that we already know about in RecordMakers.
+   *
+   * During binlog reading, if we encounter an unknown table, then we know that that table will
+   * need to be snapshotted. But, we could encounter a table that has been newly added to the
+   * config that we already have schema information for. We will still need to snapshot those
+   * tables so we are calculating that set of tables here.
+   *
+   * @param offsets the previous offset information.
+   * @param recordMakers current RecordMakers.
+   * @return
+   */
+    private Set<TableId> calculateKnownNewTables(Map<String, ?> offsets, RecordMakers recordMakers) {
+
+        if((Boolean) offsets.get(SourceInfo.HAS_FILTER_INFO)) {
+
+            Configuration oldPartialConfig = Configuration.from(offsets);
+            // note: presently ignoring config for built-in tables.  If config switched from on to off, those tables
+            // /may/ not be snapshotted.
+            Filters oldFilters = new Filters(oldPartialConfig);
+
+            Configuration currentConfig = taskContext.config();
+            Filters newFilters = new Filters(currentConfig);
+
+            Set<TableId> newKnownTables = new HashSet<>();
+            for (TableId tableId : recordMakers.getAllTableIds()) {
+                if (!oldFilters.tableFilter().test(tableId) && newFilters.tableFilter().test(tableId)){
```

Comment:
How this will work when there is a new table created during connector downtime that was already covered by old filter?

Reply:
I changed this logic a bit, so now I'm only determining what needs to be snapshotted from the config, at startup. If a new table is added during connector downtime it should not be snapshotted if it was covered by the old filter.